# Probability Analysis 
- Understanding likelihood of events. 
- What is the probability that a customer has high insurance charges?
- What is the probability that a customer is a smoker? 
- What is the probability that a high-cost customer is a smoker? 
- What is the probability that a high-cost customer is obese (high BMI)?

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

df = pd.read_csv("data\insurance.csv")
df.head()

<>:5: SyntaxWarning: invalid escape sequence '\i'
<>:5: SyntaxWarning: invalid escape sequence '\i'
C:\Users\akhia\AppData\Local\Temp\ipykernel_28664\2459740158.py:5: SyntaxWarning: invalid escape sequence '\i'
  df = pd.read_csv("data\insurance.csv")


age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

1) Probability that a customer has high insurance charges

In [2]:
threshold = df['charges'].quantile(0.9)
high_cost = df[df['charges'] >= threshold]
p_high_cost_charges = len(high_cost)/len(df)*100
p_high_cost_charges

10.014947683109119

2) Probability that a customer is a smoker

In [10]:
(df['smoker'].value_counts(normalize=True)*100)

smoker
no     79.521674
yes    20.478326
Name: proportion, dtype: float64


3) Probability that a high-cost customer is a smoker

In [13]:
p_high_cost = (high_cost['smoker'].value_counts(normalize=True)*100)
p_high_cost


smoker
yes    97.761194
no      2.238806
Name: proportion, dtype: float64

4) Probability that a high-cost customer is obese (BMI ≥ 30)

In [18]:
high_cost_obese = high_cost[high_cost['bmi'] >=30]
p_cost_obese = len(high_cost_obese)/len(high_cost)*100
p_cost_obese

96.26865671641791

## REPORT
Probability analysis shows that only about 10% of customers fall into the high-cost category. However, among these high-cost customers, nearly all are smokers (97.76%) and obese (96.27%). This indicates that smoking and high BMI are the most critical risk factors driving insurance charges.